app/bundles/ReportBundle/Model/ReportModel.php

```diff
@@ -225,13 +227,13 @@ protected function dispatchEvent($action, &$entity, $isNew = false, Event $event
      */
     public function buildAvailableReports($context)
     {
-        static $data = [];
-
-        if (empty($data[$context])) {
+        if (empty($this->reportBuilderData[$context])) {
             // Check to see if all has been obtained
-            if (isset($data['all'])) {
-                $data[$context]['tables'] = &$data['all']['tables'][$context];
-                $data[$context]['graphs'] = &$data['all']['graphs'][$context];
+            if (isset($this->reportBuilderData['all'])) {
+                $contextArray                                = explode('.', $context);
+                $alias                                       = array_pop($contextArray);
+                $this->reportBuilderData[$context]['tables'] = ($this->reportBuilderData['all']['tables'][$alias]) ?: [];
+                $this->reportBuilderData[$context]['graphs'] = (isset($this->reportBuilderData['all']['graphs'][$alias])) ? $this->reportBuilderData['all']['graphs'][$alias] : [];
```

Comment:
Shorter syntax since PHP 7:
```suggestion
                $this->reportBuilderData[$context]['graphs'] = $this->reportBuilderData['all']['graphs'][$alias] ?? [];
```